overlord/snapstate/handlers.go

```diff
@@ -777,6 +785,165 @@ func (m *SnapManager) doDownloadSnap(t *state.Task, tomb *tomb.Tomb) error {
 	return nil
 }
 
+func waitForPreDownload(task *state.Task, snapsup *SnapSetup) error {
+	st := task.State()
+	st.Lock()
+	defer st.Unlock()
+
+	tasks, err := findTasksMatchingKindAndSnap(st, "pre-download-snap", snapsup.InstanceName(), snapsup.Revision())
+	if err != nil {
+		return err
+	}
+
+	// if there is a pre-download task for the same snap, wait for it to finish
+	for _, preTask := range tasks {
+		switch preTask.Status() {
```

Comment:
(super nitpick) slightly usual to use a "switch/case" here, any reason to not just use `if` ?
```go
@@ -797,8 +797,7 @@ func waitForPreDownload(task *state.Task, snapsup *SnapSetup) error {
 
        // if there is a pre-download task for the same snap, wait for it to finish
        for _, preTask := range tasks {
-               switch preTask.Status() {
-               case state.DoingStatus:
+               if preTask.Status() == state.DoingStatus {
                        var taskIDs []string
                        if err := preTask.Get("waiting-tasks", &taskIDs); err != nil && !errors.Is(err, &state.NoStateError{}) {
                                return err
```